hudi-common/src/main/java/org/apache/hudi/common/table/read/PartialUpdateHandler.java

```diff
@@ -67,21 +67,17 @@ BufferedRecord<T> partialMerge(BufferedRecord<T> newRecord,
                                  boolean keepOldMetadataColumns) {
     // Note that: When either newRecord or oldRecord is a delete record,
     //            skip partial update since delete records do not have meaningful columns.
-    if (partialUpdateMode == PartialUpdateMode.NONE
-        || null == oldRecord
+    if (null == oldRecord
```

Comment:
do we need to check if partial update mode is null?

Reply:
nope.
the PartialUpdateMode is optional when we reach BufferedRecordMerger. 
and by the time we reach `PartialUpdateHandler`, it has to be non empty.